src/ln/msgs.rs

```diff
@@ -563,6 +565,18 @@ pub trait RoutingMessageHandler : Send + Sync {
 	fn handle_htlc_fail_channel_update(&self, update: &HTLCFailChannelUpdate);
 }
 
+///A trait to describe the functions required to access the local network topology
+pub trait NetworkTopology : Send + Sync {
+	///This should return a vec of all possible channel announcements or an error
+	fn get_all_channel_announcements(&self)-> Result<Vec<ChannelAnnouncement>, std::fmt::Error>;
```

Comment:
Hmm, this is pretty DoS-y...someone can just request the set of channel announcements and then we'll end up with a massive send buffer that they can drain (or not) at their leasure. One alternative may be to turn it into a chunked API where it returns channels from id X to Y and then we can keep track of "peer A has requested all channels, but has only gotten X to Y, so updates for anything not in that range should be dropped until they get it".

Reply:
I dont think I quite follow? Are you talking about keeping track of which channels/nodes we let them know about? Because we have to let them know about everything we know of. But this will only happen once, and only on init? If said node wants to get all information he has to shutdown and resent an init message with sync flag?

Reply:
When a node connects to us, and requests a routing table dump, PeerManager will call this and then have a full copy of all the channel announcements in memory for that peer to queue it up to send, but its of nontrivial size, so if someone creates a bunch of connections requesting a routing table dump they can make us use a lot of memory.

Reply:
I added a hashmap to keep track of to which node we send sync info to, so a public key can only get sync info once. But I suspect we might want to change that to something  more "set in stone" than a pub key. 
We might want to look at adding a timestamp to clear that entries.